src/wp-includes/block-template-utils.php

```diff
@@ -1600,12 +1622,21 @@ function inject_ignored_hooked_blocks_metadata_attributes( $changes, $deprecated
 		$post->post_author = get_current_user_id();
 	}
 
+	$wp_post = new WP_Post( $post );
+
 	if ( 'wp_template_part' === $post->post_type && ! isset( $terms['wp_template_part_area'] ) ) {
 		$area_terms                     = get_the_terms( $changes->ID, 'wp_template_part_area' );
 		$terms['wp_template_part_area'] = ! is_wp_error( $area_terms ) && ! empty( $area_terms ) ? $area_terms[0]->name : null;
+
+		/**
+		 * Hooked blocks that are ignored from a template part as first_child or last_child
+		 * are not inserted into the template part's content because they have no parent block.
+		 * Instead, they are inserted into the postmeta.
+		*/
+		update_ignored_hooked_blocks_postmeta( $wp_post );
```

Comment:
When its the first time the template is being saved we won't have a `ID` here so this solution is currently flawed too. Will need to find a different filter for `update_ignored_hooked_blocks_postmeta`. It works for the navigation because when you setup WP it creates a `wp_navigation` post.

Reply:
Right 🤔 Looking at this issue, I think it is largely an instance of the wider problem of setting metadata for a post that doesn't exist yet. WordPress' solution to this is the `meta_input` (and `tax_input`) technique that we've seen while working e.g. on https://github.com/WordPress/wordpress-develop/pull/6278.

I'm optimistic enough that we don't need to attach `update_ignored_hooked_blocks_postmeta` to a different filter (TBH I still have PTSD from all the changes we needed to make to move it from the `rest_insert_` _action_ to the `rest_pre_insert_` filter 😅 )

Instead, can we try setting the `stdClass` object's `meta_input` (instead of [calling](https://github.com/WordPress/wordpress-develop/blob/9c022260d6d0d4d0bbb2b2bc275df4e18a14044a/src/wp-includes/blocks.php#L1111) `update_post_meta`) inside `update_ignored_hooked_blocks_postmeta`?

We might need some other tweaks to `update_ignored_hooked_blocks_postmeta` for this to work, one of the bigger ones being what we [pass as context](https://github.com/WordPress/wordpress-develop/blob/9c022260d6d0d4d0bbb2b2bc275df4e18a14044a/src/wp-includes/blocks.php#L1098) to `apply_block_hooks_to_content` (instead of `get_post( $post->ID )`) -- but that's something we [needed to do anyway](https://github.com/WordPress/wordpress-develop/pull/6604/files#r1611944355) 😉 

---

(There's a chance that we _might_ want to tackle these changes in a separate PR, as they're somewhat self-contained, and arguably a prerequisite for PR. Up to you though! 😄 )

Reply:
I had completely forgotten about that method of updating post meta information. This is a much better solution than what is currently in place too. I'll create a separate PR to address this aspect of the change.

Reply:
https://github.com/WordPress/wordpress-develop/pull/6885